### PR TITLE
Save status

### LINKED ISSUE #26: Save status on shutdown, load on startup
Bot status should be made persistent through restarts

### PR DESCRIPTION
**Pull request by NicoleP23**
# Description

Changed the status to save the status it had before shutting down. If there wasn't a status before shut down, set status to "Raider Up!"

## Issues

Closes #26 

## Type of change

Select one or more of the following:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other (describe below)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. List any edge cases you tested.
If you come up with any edge cases you didn't test while writing this PR, cancel the PR and test again.

- [x]  Changed the status to something before shutting down, restart bot and see if it's there
       -status test
       -restart

- [x] Changed status to "none" and restart, should show "Raider Up!"
       -status none
       -restart

# Checklist:

- [x] All local commits have been pushed to remote
- [x] All changes on the base branch have been merged into this branch, either by rebase or merge
- [x] My code is [PEP-8](https://pep8.org/) compliant (excluding maximum line length, keep that to 100ish characters)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added [Google Docstrings](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) for all new functions/methods
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: Cogs/AdminCommands.py [hunk 1]
```diff
@@ -84,13 +84,20 @@ async def status(self, ctx, *, status):
             status (str): Text to be displayed
         """
 
+        # open a file to store the status in
+        statusFile = open('assets/status.txt', 'w')
```
Comment (CodingPenguin1) [inline Cogs/AdminCommands.py:88]: This needs to be done with `async with aiofiles.open`... Synchronous file I/O in async function breaks things

File: Cogs/AdminCommands.py [hunk 2]
```diff
@@ -84,13 +84,20 @@ async def status(self, ctx, *, status):
...
         if status.lower() == 'none':
             await self.bot.change_presence(activity=None)
             await log(self.bot, f'{ctx.author} disabled the custom status')
+            statusFile.write('Raider Up!') # Default status for when the bot restarts
         elif len(status) <= 128:
             await self.bot.change_presence(activity=discord.Game(status))
             await log(self.bot, f'{ctx.author} changed the custom status to "Playing {status}"')
+            statusFile.write(status) # write the new status to the file
+        
+        statusFile.close() # close the file
```
Comment (CodingPenguin1) [inline Cogs/AdminCommands.py:100]: If done with `with`, this is not needed

Comment (CodingPenguin1) [inline Cogs/AdminCommands.py:100]: If done with `with`, this is not needed

File: bot.py [hunk 3]
```diff
@@ -50,7 +50,13 @@ async def on_ready():
                 pass
 
     # Show the bot as online
-    await bot.change_presence(activity=discord.Game('Raider Up!'), status=discord.Status.online, afk=False)
+    # If the bot had a status prior to shutting down, restore it
+    # if it didn't, set it to 'Raider Up!'
+    statusFile = open('assets/status.txt', 'r')
```
Comment (CodingPenguin1) [inline bot.py:55]: What if the file doesn't exist?

Comment (CodingPenguin1) [reply bot.py:55]: Also aio files

File: bot.py [hunk 4]
```diff
@@ -50,7 +50,13 @@ async def on_ready():
...
+    if statusFile.mode == 'r':
```
Comment (CodingPenguin1) [inline bot.py:56]: It's `'r'` because you specified to open it with `'r'`

File: Cogs/AdminCommands.py (hunk 1 above)
Comment (CodingPenguin1) [reply Cogs/AdminCommands.py:88]: Look in `log()` function for example of how to do this

**Review by CodingPenguin1 — CHANGES_REQUESTED** [review]
Functional, some formatting issues, but mostly watch for sync ops in async. Minor thing, but look into [PEP8 variable naming conventions](https://peps.python.org/pep-0008/#function-and-variable-names)

File: bot.py [hunk 5]
```diff
@@ -50,7 +50,18 @@ async def on_ready():
                 pass
 
     # Show the bot as online
-    await bot.change_presence(activity=discord.Game('Raider Up!'), status=discord.Status.online, afk=False)
+    # If the bot had a status prior to shutting down, restore it
```
Comment (CodingPenguin1) [inline bot.py:53]: I really like this solution

**Review by CodingPenguin1 — APPROVED** [review]
